docs(gui): update the multiple session feature for Kong Manager. (#8313)
Kong/docs.konghq.com: app/_src/gateway/kong-manager/auth/sessions.md

{% endif_version %}

{% if_version gte:3.4.x %}

Attribute | Description
----------|------------

> `cookie_name` | A name for the cookie. <br> For example, `"cookie_name":"kong_cookie"`

Reviewer: ```suggestion
`cookie_name` | A name for the cookie. <br> For example, `"kong_cookie"`
```

Author: Most of this section copied from https://github.com/Kong/docs.konghq.com/blob/4ba9fc4983197291bdfed52f97f1f65a334fa523/app/_src/gateway/kong-manager/auth/sessions.md?plain=1#L88-L94. We should keep the same as the above.

`secret` | The secret used in keyed HMAC generation. Although the Session plugin's default is a random string, the `secret` _must_ be manually set for use with Kong Manager since it must be the same across all Kong workers/nodes.

> `storage` | The location where session data is stored. <br> The default value is `cookie`. It may be more secure if set to `kong`, since access to the database would be required.

Reviewer: ```suggestion
`storage` | The location where session data is stored. <br> The default value is `"cookie"`. It may be more secure if set to `"kong"`, since access to the database would be required.
```

`cookie_lifetime` | The duration (in seconds) that the session will remain open. <br> The default value is `3600`.
`cookie_secure` | Applies the Secure directive so that the cookie may be sent to the server only with an encrypted request over the HTTPS protocol. See [Session Security](#session-security) for exceptions. <br> The default value is `true`.
`cookie_samesite`| Determines whether and how a cookie may be sent with cross-site requests. See [Session Security](#session-security) for exceptions. <br> The default value is `strict`.
`hash_subject`| Whether to hash or not the subject when `store_metadata` is enabled. The default value is false.

> `store_metadata`| Whether to also store metadata of sessions, such as collecting data of sessions for a specific audience belonging to a specific subject. The default value is false. If enable this option. You should set the `storage` to `kong`.

Reviewer: ```suggestion
`store_metadata`| Whether to also store metadata of sessions, such as collecting data of sessions for a specific audience belonging to a specific subject. The default value is `false`. Upon enabling this option, please also set `storage` to `"kong"`.
```

For detailed descriptions of each configuration property, learn more in the
[Session plugin documentation](/hub/kong-inc/session).

{% if_version gte:3.10.x %}

## Multiple session management

> Sessions are isolated by default. Sessions but the current one will not be invalidated upon password updates if `store_metadata` is set to `false` or `storage` is set to `"cookie"`. In these cases, the session metadata is not stored on the server side and the session cookie is kept on the client side.

Reviewer 2: ```suggestion
When you have multiple sessions running, sessions are isolated by default.  However, the current session won't be invalidated upon password updates if `store_metadata` is set to `false` or  if `storage` is set to `"cookie"`. In these cases, the session metadata is not stored on the server side and the session cookie is kept on the client side.
```

> With `storage` set to `"kong"` and `store_metadata` set to `true`, session metadata will be stored on the server side, and all associated sessions will be invalidated upon password updates.

Reviewer 2: ```suggestion
* **Invalidate current sessions on password update**: Set `storage` to `"kong"` and set `store_metadata` to `true`. In this case, session metadata will be stored on the server side, and all associated sessions will be invalidated upon password updates.
```

    "cookie_secure":false
}
```

{% if_version gte:3.4.x %}

> If you want to invalidate all sessions when the password is changed, the following configuration could be used for Basic Auth:

Reviewer: ```suggestion
If you want to revoke all sessions when the password is changed, the following configuration could be used with the Basic Auth plugin:
```

```
enforce_rbac = on
admin_gui_auth = basic-auth
admin_gui_session_conf = {
    "cookie_name":"kong_cookie",
    "secret":"change-this-secret",
    "storage":"kong",
    "cookie_secure":false,

>     "store_metadata": true

Reviewer: ```suggestion
    "store_metadata":true
```

}
```

> If enabling the `store_metadata` and for PII reasons, the following configuration could be used instead.

Reviewer: ```suggestion
If you need to avoid storing session subjects as clear text due to PII concerns with store_metadata enabled, the following configuration can be used instead:
```

```
enforce_rbac = on
admin_gui_auth = basic-auth
admin_gui_session_conf = {
    "cookie_name":"kong_cookie",
    "secret":"change-this-secret",
    "storage":"kong",
    "cookie_secure":false,

>     "hash_subject": true,
>     "store_metadata": true

Reviewer: Let's keep the whitespace usages consistent:

```suggestion
    "hash_subject":true,
    "store_metadata":true
```
